src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,157 @@
+package duke.util;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+
+/**
+ * duke.util.Parser class that contains static methods to parse raw text.
+ */
+public class Parser {
+
+    /**
+     * Parses raw user input. Returns a Map of keywords to values, where the '/' character is
+     * parsed as a keyword.
+     *
+     * @param text The user input to be parsed
+     * @return Map of string to string, of keyword to value pairs.
+     */
+    public static Map<String, String> parseCommand(String text) {
+        HashMap<String, String> inputMap = new HashMap<>();
+        String[] splitted = text.split("\\s");
+
+        String key = splitted[0];
+        StringBuilder arguments = new StringBuilder();
+        for (int i = 1;
+             i < splitted.length;
+             i++) {
+            if (splitted[i].startsWith("/") && ValidParams.isValid(splitted[i])) {
+                if (arguments.length() > 0) {
+                    arguments.deleteCharAt(arguments.length() - 1);
+                    inputMap.put(key,
+                            arguments.toString());
+                    arguments.setLength(0);
+                } else {
+                    inputMap.put(key,
+                            null);
+                }
+                key = splitted[i];
+            } else {
+                arguments.append(String.format("%s ",
+                        splitted[i]));
+            }
+        }
+        if (arguments.length() > 0) {
+            arguments.deleteCharAt(arguments.length() - 1);
+            inputMap.put(key,
+                    arguments.toString());
+        } else {
+            if (!inputMap.containsKey(key)) {
+                inputMap.put(key,
+                        null);
+            }
+        }
+        return inputMap;
+    }
+
+    /**
+     * Tries to parse an int.
+     *
+     * @param text A supposed numeric string.
+     * @return Optional of the parsed string. If the string is erroneous,
+     * returns an empty optional.
+     */
+    public static Optional<Integer> parseInt(String text) {
+        try {
+            return Optional.of(Integer.parseInt(text));
+        } catch (NumberFormatException e) {
+            return Optional.empty();
+        }
+    }
+
+
+    /**
+     * Ensures only the right commands are parsed with ValidParams enum.
+     */
+    public enum ValidParams {
+        AT("/at"), BY("/by");
+
+        private final String value;
+
+        ValidParams(String value) {
+            this.value = value;
+        }
+
+        public static boolean isValid(String test) {
+            return Arrays.stream(ValidParams.values())
+                    .anyMatch(x -> x.value.equals(test));
```

Comment:
Could consider aligning `anyMatch` with `stream`?

Reply:
dont see much of an issue since there are 8 spaces for the wrapped line

```java
return Arrays
        .stream(...)
        .anyMatch()
```
looks rather awkward as well since the return line is rather short